packages/@react-aria/dnd/docs/useDrop.mdx

```diff
@@ -0,0 +1,436 @@
+
+```tsx example
+function DropTarget() {
+  let [file, setFile] = React.useState(null);
+  let ref = React.useRef();
+  let {dropProps, isDropTarget} = useDrop({
+    ref,
+    /*- begin highlight -*/
+    getDropOperation(types, allowedOperations) {
+      return types.has('image/png') ? 'copy' : 'cancel';
+    },
+    /*- end highlight -*/
+    async onDrop(e) {
+      let item = e.items.find(item => item.kind === 'file' && item.type === 'image/png');
+      if (item) {
+        setFile(URL.createObjectURL(await item.getFile()));
+      }
+    }
+  });
+
+  // ...
+///- begin collapse -///
+  return (
+    <div {...dropProps} role="button" tabIndex={0} ref={ref} className={`droppable ${isDropTarget ? 'target' : ''}`}>
+      {file ? <img src={file} style={{width: '100%', height: '100%', objectFit: 'contain'}} /> : 'Drop image here'}
+    </div>
+  );
+///- end collapse -///
+}
+```
+
+### onDrop
+
+The `onDrop` event also includes the `dropOperation`. This can be used to perform different actions accordingly, for example, when communicating with a backend API.
+
+```tsx
+function DropTarget(props) {
+  let ref = React.useRef();
+  let {dropProps, isDropTarget} = useDrop({
+    ref,
+    async onDrop(e) {
+      let item = e.items.find(item => item.kind === 'text' && item.types.has('my-app-file'));
+      if (!item) {
+        return;
+      }
+
+      let data = JSON.parse(await item.getText('my-app-file'));
+      /*- begin highlight -*/
+      switch (e.dropOperation) {
+        case 'move':
+          MyAppFileService.move(data.filePath, props.filePath);
+          break;
+        case 'copy':
+          MyAppFileService.copy(data.filePath, props.filePath);
+          break;
+        case 'link':
+          MyAppFileService.link(data.filePath, props.filePath);
+          break;
+      }
+      /*- end highlight -*/
+    }
+  });
+
+  // ...
+}
+```
+
+## Events
+
+Drop targets receive a number of events during a drag session. These are:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={Object.fromEntries(Object.entries(docs.links[docs.exports.useDrop.parameters[0].value.id].properties).filter(([k]) => k.startsWith('on')))} />
+</TypeContext.Provider>
+
+This example logs all events that occur within the drop target:
+
+```tsx example
+function DropTarget() {
+  let [events, setEvents] = React.useState([]);
+  let onEvent = e => setEvents(events => [JSON.stringify(e), ...events]);
+  let ref = React.useRef();
+  let {dropProps, isDropTarget} = useDrop({
+    ref,
+    onDropEnter: onEvent,
+    onDropMove: onEvent,
+    onDropActivate: onEvent,
```

Comment:
We'll have to remove this from the example since we are making this prop private